<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>436</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sudoku OCR</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>9</number>
    </property>
    <property name="topMargin">
     <number>9</number>
    </property>
    <property name="rightMargin">
     <number>9</number>
    </property>
    <property name="bottomMargin">
     <number>9</number>
    </property>
    <item row="0" column="0">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="page_0">
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="topMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <property name="bottomMargin">
         <number>9</number>
        </property>
        <item row="0" column="0" alignment="Qt::AlignVCenter">
         <widget class="QTextBrowser" name="textBrowser">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>500</width>
            <height>150</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="html">
           <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;                                                &lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;                                                 &lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;                                                &lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;SUDOKU OCR&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;                                                 &lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Solve any Sudoku! Steps:&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;-Load a Sudoku&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;-Let magic processing run...&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;-Correct rotation and digits if necessary&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;-The Sudoku is solved!&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="ImgSelectorButton" name="ImageSelectorButton">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>100</height>
           </size>
          </property>
          <property name="text">
           <string>Drag image or click to select from disk</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_1">
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="backButton_1">
            <property name="font">
             <font>
              <family>Liberation Serif</family>
              <italic>false</italic>
              <bold>false</bold>
              <kerning>true</kerning>
             </font>
            </property>
            <property name="text">
             <string>↩️ Back</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_1">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="validateButton_1">
            <property name="text">
             <string>Continue ✅</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="0">
         <widget class="ImageDisplay" name="imgDisplay_1">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_2">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="ImageDisplay" name="imgDisplay_2">
          <property name="text">
           <string>ImageDisplay</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_3">
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="ShapeDefiner" name="ShapeDefiner_3">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="text">
           <string>ShapeDefiner</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="validateButton_3">
            <property name="text">
             <string>Continue ✅</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_4">
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="0" column="0" colspan="3">
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="ImageDisplay" name="imgDisplay_4">
             <property name="text">
              <string>ImageDisplay</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QGridLayout" name="gridLayout_4_" columnstretch="0,0,0,0,0,0,0,0,0">
             <item row="1" column="3">
              <widget class="QSpinBox" name="spinBox_03"/>
             </item>
             <item row="3" column="8">
              <widget class="QSpinBox" name="spinBox_26"/>
             </item>
             <item row="3" column="2">
              <widget class="QSpinBox" name="spinBox_20"/>
             </item>
             <item row="2" column="6">
              <widget class="QSpinBox" name="spinBox_15"/>
             </item>
             <item row="3" column="6">
              <widget class="QSpinBox" name="spinBox_24"/>
             </item>
             <item row="9" column="3">
              <widget class="QSpinBox" name="spinBox_75"/>
             </item>
             <item row="3" column="3">
              <widget class="QSpinBox" name="spinBox_21"/>
             </item>
             <item row="4" column="2">
              <widget class="QSpinBox" name="spinBox_29"/>
             </item>
             <item row="6" column="1">
              <widget class="QSpinBox" name="spinBox_46"/>
             </item>
             <item row="7" column="8">
              <widget class="QSpinBox" name="spinBox_62"/>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spinBox_01"/>
             </item>
             <item row="2" column="3">
              <widget class="QSpinBox" name="spinBox_12"/>
             </item>
             <item row="5" column="2">
              <widget class="QSpinBox" name="spinBox_38"/>
             </item>
             <item row="8" column="1">
              <widget class="QSpinBox" name="spinBox_64"/>
             </item>
             <item row="8" column="3">
              <widget class="QSpinBox" name="spinBox_66"/>
             </item>
             <item row="5" column="5">
              <widget class="QSpinBox" name="spinBox_41"/>
             </item>
             <item row="3" column="1">
              <widget class="QSpinBox" name="spinBox_19"/>
             </item>
             <item row="6" column="5">
              <widget class="QSpinBox" name="spinBox_50"/>
             </item>
             <item row="9" column="6">
              <widget class="QSpinBox" name="spinBox_78"/>
             </item>
             <item row="5" column="6">
              <widget class="QSpinBox" name="spinBox_42"/>
             </item>
             <item row="6" column="8">
              <widget class="QSpinBox" name="spinBox_53"/>
             </item>
             <item row="2" column="5">
              <widget class="QSpinBox" name="spinBox_14"/>
             </item>
             <item row="4" column="5">
              <widget class="QSpinBox" name="spinBox_32"/>
             </item>
             <item row="7" column="3">
              <widget class="QSpinBox" name="spinBox_57"/>
             </item>
             <item row="9" column="8">
              <widget class="QSpinBox" name="spinBox_80"/>
             </item>
             <item row="3" column="0">
              <widget class="QSpinBox" name="spinBox_18"/>
             </item>
             <item row="3" column="7">
              <widget class="QSpinBox" name="spinBox_25"/>
             </item>
             <item row="1" column="6">
              <widget class="QSpinBox" name="spinBox_06"/>
             </item>
             <item row="5" column="8">
              <widget class="QSpinBox" name="spinBox_44"/>
             </item>
             <item row="1" column="4">
              <widget class="QSpinBox" name="spinBox_04"/>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="spinBox_10"/>
             </item>
             <item row="8" column="5">
              <widget class="QSpinBox" name="spinBox_68"/>
             </item>
             <item row="2" column="4">
              <widget class="QSpinBox" name="spinBox_13"/>
             </item>
             <item row="5" column="0">
              <widget class="QSpinBox" name="spinBox_36"/>
             </item>
             <item row="7" column="7">
              <widget class="QSpinBox" name="spinBox_61"/>
             </item>
             <item row="9" column="5">
              <widget class="QSpinBox" name="spinBox_77"/>
             </item>
             <item row="4" column="6">
              <widget class="QSpinBox" name="spinBox_33"/>
             </item>
             <item row="5" column="1">
              <widget class="QSpinBox" name="spinBox_37"/>
             </item>
             <item row="6" column="2">
              <widget class="QSpinBox" name="spinBox_47"/>
             </item>
             <item row="5" column="7">
              <widget class="QSpinBox" name="spinBox_43"/>
             </item>
             <item row="2" column="0">
              <widget class="QSpinBox" name="spinBox_09"/>
             </item>
             <item row="6" column="7">
              <widget class="QSpinBox" name="spinBox_52"/>
             </item>
             <item row="8" column="4">
              <widget class="QSpinBox" name="spinBox_67"/>
             </item>
             <item row="3" column="4">
              <widget class="QSpinBox" name="spinBox_22"/>
             </item>
             <item row="7" column="1">
              <widget class="QSpinBox" name="spinBox_55"/>
             </item>
             <item row="7" column="5">
              <widget class="QSpinBox" name="spinBox_59"/>
             </item>
             <item row="1" column="2">
              <widget class="QSpinBox" name="spinBox_02"/>
             </item>
             <item row="8" column="0">
              <widget class="QSpinBox" name="spinBox_63"/>
             </item>
             <item row="1" column="7">
              <widget class="QSpinBox" name="spinBox_07"/>
             </item>
             <item row="2" column="8">
              <widget class="QSpinBox" name="spinBox_17"/>
             </item>
             <item row="7" column="4">
              <widget class="QSpinBox" name="spinBox_58"/>
             </item>
             <item row="7" column="0">
              <widget class="QSpinBox" name="spinBox_54"/>
             </item>
             <item row="1" column="0">
              <widget class="QSpinBox" name="spinBox_00"/>
             </item>
             <item row="9" column="7">
              <widget class="QSpinBox" name="spinBox_79"/>
             </item>
             <item row="8" column="8">
              <widget class="QSpinBox" name="spinBox_71"/>
             </item>
             <item row="2" column="2">
              <widget class="QSpinBox" name="spinBox_11"/>
             </item>
             <item row="4" column="1">
              <widget class="QSpinBox" name="spinBox_28"/>
             </item>
             <item row="8" column="2">
              <widget class="QSpinBox" name="spinBox_65"/>
             </item>
             <item row="7" column="6">
              <widget class="QSpinBox" name="spinBox_60"/>
             </item>
             <item row="6" column="3">
              <widget class="QSpinBox" name="spinBox_48"/>
             </item>
             <item row="5" column="3">
              <widget class="QSpinBox" name="spinBox_39"/>
             </item>
             <item row="4" column="3">
              <widget class="QSpinBox" name="spinBox_30"/>
             </item>
             <item row="4" column="8">
              <widget class="QSpinBox" name="spinBox_35"/>
             </item>
             <item row="5" column="4">
              <widget class="QSpinBox" name="spinBox_40"/>
             </item>
             <item row="3" column="5">
              <widget class="QSpinBox" name="spinBox_23"/>
             </item>
             <item row="4" column="0">
              <widget class="QSpinBox" name="spinBox_27"/>
             </item>
             <item row="2" column="7">
              <widget class="QSpinBox" name="spinBox_16"/>
             </item>
             <item row="8" column="7">
              <widget class="QSpinBox" name="spinBox_70"/>
             </item>
             <item row="1" column="8">
              <widget class="QSpinBox" name="spinBox_08"/>
             </item>
             <item row="6" column="4">
              <widget class="QSpinBox" name="spinBox_49"/>
             </item>
             <item row="6" column="6">
              <widget class="QSpinBox" name="spinBox_51"/>
             </item>
             <item row="4" column="7">
              <widget class="QSpinBox" name="spinBox_34"/>
             </item>
             <item row="7" column="2">
              <widget class="QSpinBox" name="spinBox_56"/>
             </item>
             <item row="9" column="4">
              <widget class="QSpinBox" name="spinBox_76"/>
             </item>
             <item row="8" column="6">
              <widget class="QSpinBox" name="spinBox_69"/>
             </item>
             <item row="9" column="0">
              <widget class="QSpinBox" name="spinBox_72"/>
             </item>
             <item row="9" column="1">
              <widget class="QSpinBox" name="spinBox_73"/>
             </item>
             <item row="6" column="0">
              <widget class="QSpinBox" name="spinBox_45"/>
             </item>
             <item row="9" column="2">
              <widget class="QSpinBox" name="spinBox_74"/>
             </item>
             <item row="4" column="4">
              <widget class="QSpinBox" name="spinBox_31"/>
             </item>
             <item row="1" column="5">
              <widget class="QSpinBox" name="spinBox_05"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="validateButton_4">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Continue ✅</string>
            </property>
            <property name="checkable">
             <bool>false</bool>
            </property>
            <property name="default">
             <bool>false</bool>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>700</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ImgSelectorButton</class>
   <extends>QPushButton</extends>
   <header>GUI/ImgSelectorButton.h</header>
  </customwidget>
  <customwidget>
   <class>ImageDisplay</class>
   <extends>QLabel</extends>
   <header>GUI/ImageDisplay.h</header>
  </customwidget>
  <customwidget>
   <class>ShapeDefiner</class>
   <extends>QLabel</extends>
   <header>GUI/ShapeDefiner.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
